### PR TITLE
[intfstat] add intfstat script to output RIF counters

### PR DESCRIPTION
**Pull request by mykolaf**
* add new intfstat script
* introduce utilities-common
* move some functions to calculate network statistics to utilities-common
* 'add show interfaces counters rif' CLI
* remove 'show interface counters --clear option
* introduce mock_tables under sonic-utilities-tests, allowing to unittest the CLIs, check the command output based on mocked values.
* add unit test for intfstat
Signed-off-by: Mykola Faryma <mykolaf@mellanox.com>

**- What I did**
Add a script to output RIF counters.
**- How I did it**

**- How to verify it**

**- Previous command output (if the output of a command-line utility has changed)**
none
**- New command output (if the output of a command-line utility has changed)**
```
      IFACE       RX_OK    RX_BPS    RX_PPS    RX_ERR    TX_OK    TX_BPS    Tx_PPS    TX_ERR
-----------       -------  --------  ----------------  -------  --------  ---------  --------  
Ethernet0         2180       N/A       N/A        0       44697      N/A       N/A     0      
Ethernet4         47342      N/A       N/A        0       8362       N/A       N/A     0      
Ethernet8         2188       N/A       N/A        0       9379       N/A       N/A     0      
Ethernet12        46882      N/A       N/A        0       44448      N/A       N/A     0      
 ...
Portchannel0002   2833       N/A       N/A        0       0          N/A       N/A     0   
 ...
Vlan2             2833       N/A       N/A        0       4170       N/A       N/A     0
```
-->

### CODE REVIEW

**Comment by mykolaf** [comment]
Hi @lguohan , I added the mock redis for unittests, like we have for sonic-snmpagent.
Can we install mockredispy==2.9.3 in the build environment?

**Comment by mykolaf** [comment]
Does this vs-test run 'python setup.py test'? In that case I expect it to install 'mock' and  'mockredispy' packages. What does build_sonic_utilities.sh do? 

@lguohan  I added mock and mockredispy as test dependencies to the setuptools, still can't get it to work.

**Comment by lguohan** [comment]
retest this please

**Comment by lguohan** [comment]
retest this please

**Comment by lguohan** [comment]
retest this please

File: show/main.py [hunk 1]
```diff
@@ -554,23 +554,41 @@ def status(interfacename, verbose):
...
+                cmd += " -p {}".format(period)
+
+        run_command(cmd, display_cmd=verbose)
+
+# 'counters' subcommand ("show interfaces counters rif")
+@counters.command()
+@click.argument('interface', metavar='<interface_name>', required=False, type=str)
+@click.option('-p', '--period')
+@click.option('--verbose', is_flag=True, help="Enable verbose output")
+def rif(interface, period, verbose):
```
Comment (jleveque) [inline show/main.py]: Please add support for interface naming mode.

**Comment by mykolaf** [comment]
retest this please

**Comment by mykolaf** [comment]
retest this please

File: show/main.py (hunk 1 above)
Comment (liat-grozovik) [reply show/main.py]: @jleveque can you please review and confirm the recent changes apply to your comments?

Comment (mykolaf) [reply show/main.py]: @jleveque BTW, there is something wrong with the build, the issue is common for pr-build and regular build.

**Comment by jleveque** [comment]
Retest this please.

File: show/main.py (hunk 1 above)
Comment (jleveque) [reply show/main.py]: @mykolaf: Thanks for the heads-up. I fixed a couple issues with the check builds, and I'm looking into the remaining issues.

**Comment by mykolaf** [comment]
retest this please

**Comment by mykolaf** [comment]
retest this please

File: show/main.py (hunk 1 above)
Comment (mykolaf) [reply show/main.py]: @jleveque I think this one is resolved

**Pull request merged**